helix-view/src/document.rs

```diff
@@ -138,14 +145,6 @@ pub fn normalize_path(path: &Path) -> PathBuf {
     ret
 }
 
-// Returns the canonical, absolute form of a path with all intermediate components normalized.
-//
-// This function is used instead of `std::fs::canonicalize` because we don't want to verify
-// here if the path exists, just normalize it's components.
-pub fn canonicalize_path(path: &Path) -> std::io::Result<PathBuf> {
-    std::env::current_dir().map(|current_dir| normalize_path(&current_dir.join(path)))
```

Comment:
You still want this because paths need to be converted to absolute. Not taking `current_dir()` into account will consider `myfile.c` and `/full/path/to/myfile.c` as two separate files.

Reply:
Tilde expansion should be a step before normalization: https://github.com/netvl/shellexpand/blob/e707d07a0a2638df7ab26b152f814e80ce3b3992/src/lib.rs#L665-L690